processing/src/main/java/io/druid/query/ResultGranularTimestampComparator.java

```diff
@@ -29,18 +29,23 @@
 public class ResultGranularTimestampComparator<T> implements Comparator<Result<T>>
 {
   private final QueryGranularity gran;
+  private final boolean descending;
 
-  public ResultGranularTimestampComparator(QueryGranularity granularity)
+  public ResultGranularTimestampComparator(QueryGranularity granularity, boolean descending)
```

Comment:
Ok, I'll make a factory method in ResultGranularTimestampComparator to handle that.